### PR TITLE
gpg-suite: fix broken uninstall logic

### PR DESCRIPTION
**Pull request by gromgit**
Doesn't make sense to check a link _after_ it's removed. :)

Fixes https://github.com/orgs/Homebrew/discussions/3402.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by carlocab** [comment]
Looks like the other `gpg-suite-*` casks need fixing too.

**Comment by gromgit** [comment]
> Looks like the other `gpg-suite-*` casks need fixing too.

I'm on it.

Side note: This seems to be a result of a b0rken `brew style` fix:
```
$ brew style --fix --cask gpg-suite
Casks/gpg-suite.rb:27:21: W: [Corrected] Remove unnecessary existence checks File.exist?.
      File.rm(exec) if File.exist?(exec) && File.readlink(exec).include?("MacGPG2")
                    ^^^^^^^^^^^^^^^^^^^^
Casks/gpg-suite.rb:27:28: C: [Corrected] Unnecessary spacing detected.
      FileUtils.rm_rf(exec)  && File.readlink(exec).include?("MacGPG2")
                           ^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

File: Casks/gpg-suite.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
   uninstall_postflight do
     ["gpg", "gpg2", "gpg-agent"].map { |exec_name| "/usr/local/bin/#{exec_name}" }.each do |exec|
-      FileUtils.rm_rf(exec) && File.readlink(exec).include?("MacGPG2")
+      File.readlink(exec).include?("MacGPG2") && FileUtils.rm_rf(exec)
```
Comment (carlocab) [inline Casks/gpg-suite.rb:27]:
```suggestion
      next unless File.exist?(exec)
      next unless File.readlink(exec).include?("MacGPG2")

      FileUtils.rm_rf(exec)
```
I'm tempted to make `/usr/local/bin` a `Pathname` so we don't need to keep using the `File` class.

File: Casks/gpg-suite-no-mail.rb [hunk 2]
```diff
@@ -115,7 +115,7 @@
 
   uninstall_postflight do
     ["gpg", "gpg2", "gpg-agent"].map { |exec_name| "/usr/local/bin/#{exec_name}" }.each do |exec|
-      FileUtils.rm_rf(exec) && File.readlink(exec).include?("MacGPG2")
+      File.readlink(exec).include?("MacGPG2") && FileUtils.rm_rf(exec)
```
Comment (carlocab) [inline Casks/gpg-suite-no-mail.rb:118]: Note that the `readlink` will error if the file doesn't exist. Which is probably why we needed the `File.exist?` check in the first place.

File: Casks/gpg-suite.rb (hunk 1 above)
Comment (gromgit) [reply Casks/gpg-suite.rb:27]: So something like:
```
["gpg", "gpg2", "gpg-agent"].map { |exec_name| Pathname.new("/usr/local/bin/#{exec_name}") }.each do |exec|
    next unless exec.exist?
    next unless exec.readlink.to_s.include?("MacGPG2")

    exec.unlink
end
```
?

Comment (carlocab) [reply Casks/gpg-suite.rb:27]: Sure, or even

```ruby
["gpg", "gpg2", "gpg-agent"].map { |exec_name| Pathname("/usr/local/bin")/exec_name }.each do |exec|
    next unless exec.exist?
    next unless exec.readlink.to_s.include?("MacGPG2")

    exec.unlink
end
```

**Comment by gromgit** [comment]
@carlocab, I've gone with your suggested three-liner:
```ruby
["gpg", "gpg2", "gpg-agent"].map { |exec_name| Pathname("/usr/local/bin")/exec_name }.each do |exec|
    exec.unlink if exec.exist? && exec.readlink.to_s.include("MacGPG2")
end
```
It seems to work in my local test environment, and doesn't get mangled into illogic by `brew style`.

**Review by carlocab — APPROVED** [review]
Thanks @gromgit!

**Pull request merged**